internal/infrastructure/ceph/rbd.go

```diff
@@ -194,14 +267,31 @@ func applyDiffDataRecords(
 	diffFileReader *bufio.Reader,
 	expansionUnitSize uint64,
 	isDstBlockDevice bool,
+	rawChecksumChunkSize uint64,
+	rawChecksumFile *os.File,
 ) error {
+
+	// If the destination is a raw image and a checksum file exists
+	// (disableChecksumVerify == false), apply the diff while verifying
+	// checksums.
+	if !isDstBlockDevice && rawChecksumFile != nil {
+		return applyDiffDataRecordsWithChecksumVerification(
+			dstFile,
+			diffFileReader,
+			expansionUnitSize,
+			rawChecksumChunkSize,
+			rawChecksumFile,
+		)
+	}
+
```

Comment:
How about calling `applyDiffDataRecordsWithChecksumVerification` from `applyDiffToRawImage` instead of `applyDiffDataRecords`? Then we don't have to add new arguments to `applyDiffDataRecords`.